pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java

```diff
@@ -79,23 +79,36 @@ protected UsernamePasswordCredentials retrieveCredentials(final WebContext conte
             credentials = getCredentialsExtractor().extract(context);
             logger.debug("usernamePasswordCredentials: {}", credentials);
             if (credentials == null) {
-                String redirectionUrl = CommonHelper.addParameter(this.loginUrl, this.usernameParameter, username);
-                redirectionUrl = CommonHelper.addParameter(redirectionUrl, ERROR_PARAMETER, MISSING_FIELD_ERROR);
-                logger.debug("redirectionUrl: {}", redirectionUrl);
-                final String message = "Username and password cannot be blank -> return to the form with error";
-                logger.debug(message);
-                throw RequiresHttpAction.redirect(message, context, redirectionUrl);
+				// it's an AJAX request -> unauthorized (instead of a
```

Comment:
OK. If there is a problem, we'll fix it, but in another pull request. Just recommit the original current master file and the change will go away from the pull request. We'll deal with that problem later on. You can already open a Github issue so we don't forget about it.